lib/src/signer.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(feature = "signing-gpgme")]
+use gpgme::{Context, Key, Protocol};
+
+use crate::backend::BackendResult;
+
+pub trait Signer {
+    fn sign(&self, input: &[u8]) -> BackendResult<String>;
+}
```

Comment:
nit: I'd put this in `backend.rs` and make this file be only the implementations (or maybe even just the gpgme implementation)

Reply:
sure, will do